### PR TITLE
Prevent mapper to perform an insert and a move to the same index path

### PR DESCRIPTION
**Pull request by Heinemark**
Not the final solution, just opening the PR for discussions.

### CODE REVIEW

**Comment by Heinemark** [comment]
Was it something like this you had in mind @accatyyc? But divided into small functions ofc.

**Comment by joelekstrom** [comment]
Yeah, something like that! Extracting it a bit would be good because it feels like that function is getting complicated. Do you know if it solves the issue?

**Comment by Heinemark** [comment]
Haven't seen one single crash since I started running a version with this change, so it's looks promising.

**Comment by Heinemark** [comment]
Maybe something like this @accatyyc?

File: FRZDatabaseViewMapper.m
Comment (joelekstrom) [inline FRZDatabaseViewMapper.m]: Should this maybe also check if the old indexPath was deleted?

**Review by joelekstrom — COMMENTED** [review]
Excellent refactor 👌

File: FRZDatabaseViewMapper.m
Comment (Heinemark) [reply FRZDatabaseViewMapper.m]: True, better safe than sorry :)

File: FRZDatabaseViewMapper.m
Comment (Heinemark) [inline FRZDatabaseViewMapper.m]: Hm, when thinking about it, since this index path is obviously already among inserted index paths this call is unnecessary 🤔

Comment (joelekstrom) [reply FRZDatabaseViewMapper.m]: Yeah, I thought about it but it doesn’t really matter since the other function will discard the insert. Might be nice for clarity

Comment (Heinemark) [reply FRZDatabaseViewMapper.m]: Yeah I guess it's a bit easier to understand like this.

**Comment by Heinemark** [comment]
I can't approve a PR I created myself, but unless we also need to check if an updated index path can't be among inserted sections too , everything looks great.

**Comment by joelekstrom** [comment]
cool! updated index paths are only working on the sections before deletions, so you can't update a newly inserted section (and newly inserted cells are always updated anyway) so should be fine!

**Review by quebin — APPROVED** [review]

**Pull request merged**
